### DISCUSSION TITLE
How to detect app is running in the designer?

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#13526] [Q&A] [answered]
I notice when VS opens an AXAML file in the designer, my .exe's entrypoint is executed. It then loads data files on behalf of the user from %localappdata%, but this isn't appropriate in the designer, as I have view models with mock data. 

How can my `BuildAvaloniaApp` method detect that it's running under the designer and avoid loading these files?

I checked for environment variables and nothing unusual seems to be set, although I _could_ check for VS specific environment variables and assume I'm in the designer when they are present.
Another approach could be that, given `BuildAvaloniaApp` is invoked (directly) by the designer instead of via the `Main` method, my `Main` method could set some context.

**Comment by timunie** [comment, marked as answer]
Design.IsDesignTime

**Reply by AArnott** [reply]
Thanks!
